PagerDuty/public-support-scripts
# Tests, Logging, Docs, & (smol) Refactor

- refact: split single class into csvparsing and apicalling classes
- feat: added logging (and print of url to check account's custom fields)
- chore(style): changed logger quote type
- build: deny `.env` files and allow exception for `test.csv` files
- feat: testing for `import_custom_fields` script

**Link back to Jira ticket:**

## Description
- Added tests (along with .env functionality should one want to use it in future CI/CD)
- Added some additional Documentation
- And did a slight refactor, separating the core script into a parsing and an api-call making class.
- Added logging to main script


## Implementation
Only two gems added, one for pulling local env variables from `.env` and one for testing.
(logger is a ruby built-in)


### Steps to test changes
- local install: `bundle install --path vendor/bundle`
- <place test api-key in `.env` file>
- run tests: `bundle exec rspec spec.rb`
  - alternate: `TEST_PAGERDUTY_API_KEY='#######' bundle exec rspec spec.rb`
- manual test: `bundle exec ruby import_custom_fields.rb -k ##### -f test.csv`
- check logs: `cat application.log`

## Documentation 
- [x] There is documentation that needs to be updated upon merging these changes

Links to any documentation to be updated:
